### PR TITLE
feat: Refund command

### PR DESCRIPTION
**Pull request by adamspofford-dfinity**
Adds a refund command to interact with `swap.error_refund_icp`.

Implements SDK-563.

### CODE REVIEW

File: src/commands/mod.rs [hunk 1]
```diff
@@ -44,6 +45,8 @@ pub enum Command {
     /// can submit proposals (such as a Motion Proposal) to be voted on by other neuron
     /// holders.
     MakeProposal(make_proposal::MakeProposalOpts),
+    /// Signs a request for a refund of ICP from the swap canister.
+    Refund(refund::RefundOpts),
```
Comment (DanielThurau) [inline src/commands/mod.rs]: Nit: Refund could probably be applied to ledger transfers and neuron staking. Perhaps a more detailed name could avoid some user confusion. I recommend `RefundSwap` but open to counter suggestions.

File: src/commands/refund.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        AnyhowResult, TargetCanister,
+    },
+    SnsCanisterIds,
+};
+
+use super::transfer;
+
+/// Signs a message to request a refund from the SNS swap canister.
+/// If the swap was aborted or failed, or some of your contributed ICP never made it into a neuron,
+/// this command can retrieve your unused ICP, minus transaction fees.
```
Comment (DanielThurau) [inline src/commands/refund.rs]: Nit: Could you copy this to mod.rs as that is where clap pulls the descriptions displayed when using `sns-quill --help`

**Comment by DanielThurau** [comment]
LGTM! No real changes besides a couple nits and bumping the version. Ready to approve once those are addressed

File: src/commands/refund.rs (hunk 2 above)
Comment (adamspofford-dfinity) [reply src/commands/refund.rs]: It only does that if you provide one; if the full help message should be displayed in all circumstances then the same effect is achieved by simply leaving it blank, no?

Comment (DanielThurau) [reply src/commands/refund.rs]: I didn't know that Clap was capable of doing that routing. Leaving it blank removes comment duplication so SGTM

**Review by DanielThurau — APPROVED** [review]

File: src/commands/mod.rs [hunk 3]
```diff
@@ -44,6 +45,7 @@ pub enum Command {
     /// can submit proposals (such as a Motion Proposal) to be voted on by other neuron
     /// holders.
     MakeProposal(make_proposal::MakeProposalOpts),
+    RefundSwap(refund_swap::RefundSwapOpts),
```
Comment (max-dfinity) [inline src/commands/mod.rs]: Can you add some documentation?

File: src/commands/refund_swap.rs [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    SnsCanisterIds,
+};
+
+use super::transfer;
+
+/// Signs a message to request a refund from the SNS swap canister.
+/// If the swap was aborted or failed, or some of your contributed ICP never made it into a neuron,
+/// this command can retrieve your unused ICP, minus transaction fees.
+#[derive(Parser)]
+pub struct RefundSwapOpts {
```
Comment (max-dfinity) [inline src/commands/refund_swap.rs]: Given what the description is, maybe a better name would be "GetSwapRefund"?  "RefundSwap" sounds like an action taken on the other side of the transaction

File: src/commands/mod.rs (hunk 3 above)
Comment (adamspofford-dfinity) [reply src/commands/mod.rs]: The documentation comment is inherited from the RefundSwapOpts, as described in the previous conversation.

**Comment by adamspofford-dfinity** [comment]
@max-dfinity anything else?

**Pull request merged**